### PR TITLE
Feature/xdebug guide

### PR DESCRIPTION
**Pull request by pswaine**

### CODE REVIEW

**Comment by pswaine** [comment]
@sroze - one problem with this guide is that the dockerfile will always install php_xdebug. Could that be solved using separate dockerfiles with condition restricted build steps? Do build steps support that?

**Comment by andytson-inviqa** [comment]
docker build args could be used to control the installation. That said xdebug installation is not the main cost but runtime it is. Therefore it's more convenient to have the ini that loads it in controlled by a runtime env var instead.

**Comment by pswaine** [comment]
What would the env var controlled ini look @andytson-inviqa? Something like this?:
```
zend_extension=xdebug.so
xdebug.remote_enable=${XDEBUG_REMOTE_ENABLED}
xdebug.remote_host=${XDEBUG_REMOTE_HOST}
xdebug.remote_port=${XDEBUG_REMOTE_PORT}
```

Or is there a way to conditionalise the loading of the extension altogether?

**Comment by andytson-inviqa** [comment]
confd can do something like:
```
{{ if getenv XDEBUG_REMOTE_ENABLED }}
zend_extension=xdebug.so
xdebug.remote_enable={{ getenv XDEBUG_REMOTE_ENABLED }}
xdebug.remote_host={{ getenv XDEBUG_REMOTE_HOST }}
xdebug.remote_port={{ getenv XDEBUG_REMOTE_PORT }}
{{ end }}
```

**Comment by pswaine** [comment]
@andytson-inviqa - cheers, will give it a try

**Comment by pswaine** [comment]
@andytson-inviqa - I can't get this to work, the bracketed values are interpreted by php as string literals instead of env variables.

**Comment by andytson-inviqa** [comment]
that's as it's confd templating. See the docker base images, which do that

**Comment by pswaine** [comment]
@andytson-inviqa got that working now thanks. Any suggestions on a way to conditionally replace the 40-tideways.ini config? i.e. leave the default file in place unless it needs disabling.

**Comment by sroze** [comment]
@pswaine I'd say the same way, using `if`s and environment variables in a template file for it?

**Comment by pswaine** [comment]
@sroze - thats what I ended up doing. I was just wondering if there was a more elegant way that avoids duplicating the content of the default 40-tideways.ini in my tideways config template.

```
{{ if getenv "XDEBUG_REMOTE_ENABLED" }}
;extension=tideways.so
{{ else }}
; Configuration for Tideways Profiler Extension
; priority=40
extension=tideways.so

; Tideways Application API-Key to configure when using just one application on
; this php installation.
;tideways.api_key=

; Configure the profiling sample rate for this PHP server globally.  The given
; number is an integer representing percent between 0 and 100
tideways.sample_rate=25

; Automatically detect transactions and exceptions of a given framework The
; following frameworks are currently supported:
;
; symfony2, symfony2c, shopware, oxid, magento, zend1, zend2, laravel,
; wordpress
;tideways.framework=
{{ end }}
```

File: content/guides/remote-environment-xdebug.md [hunk 1]
```diff
@@ -0,0 +1,514 @@
...
+xdebug.remote_port=9000
+xdebug.remote_host=sshforward
+xdebug.remote_log=
+```
+
+This will enable remote debugging on port 9000, using the "sshforward" host. Details on setting up the "sshforward" host are in the next section of the guide.
+
+In addition, you will also need to create the following file in the repository at "docker/etc/php/7.1/fpm/conf.d/40-tideways.ini":
+
+> **40-tideways.ini**
```
Comment (sroze) [inline content/guides/remote-environment-xdebug.md]: Can we move that into a "If you have Tideways" section?

Comment (pswaine) [inline content/guides/remote-environment-xdebug.md]: @sroze I've split Tideways instructions into 2 additional appendices.

**Comment by pswaine** [comment]
@sroze - updated following feedback, theres now separate appendices for Tideways. Unfortunately I rebased so the changes are not apparent.

**Comment by sroze** [comment]
👌 💯

**Pull request merged**